spec/services/timeline_events/was_last_target_service_spec.rb

```diff
@@ -58,6 +58,19 @@
     end
   end
 
+  describe "#was_last_target?" do
+    let!(:submission_1) { complete_target(target_2, student) }
+
+    before do
+      submission_1.update!(archived_at: Time.zone.now)
+      team.students.each { |f| complete_target(target, f) }
+    end
+
+    it "returns false as one of the milestone submissions is archived" do
+      expect(described_class.new(submission_1).was_last_target?).to eq(false)
+    end
+  end
+
   describe "#was_last_target?" do
```

Comment:
These specs are poorly organized. `describe` and `context` are used incorrectly. Since this class has only one public method - `#was_last_target?`, there should be only one of these `describe`-s, with several contexts inside. For example, here's a starter:

```rb
describe "#was_last_target?" do
  context "when all milestone assignments have been completed" do
    before { team.students.each { |f| complete_target(target_1, f) } }
    let(:submission_1) { complete_target(target_2, student) }

    it "returns true" do
      expect(described_class.new(submission_1).was_last_target?).to eq(true)
    end
  end

  context "when only one student has completed all assignments" do
    let(:submission_1) { complete_target(target_1, student) } # <- student target
    let(:submission_2) { complete_target(target_2, student) } # <- team target
    let(:submission_3) { complete_target(target_3, student) } # <- non-milestone

    it "returns false" do
      expect(described_class.new(submission_3).was_last_target?).to eq(false)
    end
  end
end
```

This file needs to be reorganized for it to make sense for later edits.

Additional issues:

1. There is a spec with the descriptor `returns false as only one submission is reviewed` that doesn't seem to test anything new.
2. There is a context `when cohort has ended` which sets cohort as ended - but the service has nothing inside it that checks the cohort end date, so I'm not sure why it exists.

Reply:
Updated the specs;